src/bootstrap/compile.rs

```diff
@@ -512,14 +512,14 @@ impl Step for Rustc {
     }
 }
 
-pub fn rustc_cargo(builder: &Builder, cargo: &mut Command) {
+pub fn rustc_cargo(builder: &Builder, cargo: &mut Command, is_test: bool) {
     cargo.arg("--features").arg(builder.rustc_features())
          .arg("--manifest-path")
          .arg(builder.src.join("src/rustc/Cargo.toml"));
-    rustc_cargo_env(builder, cargo);
+    rustc_cargo_env(builder, cargo, is_test);
 }
 
-pub fn rustc_cargo_env(builder: &Builder, cargo: &mut Command) {
+pub fn rustc_cargo_env(builder: &Builder, cargo: &mut Command, is_test: bool) {
```

Comment:
Instead of modifying `rustc_cargo_env`, which forces one to pass a magic boolean everywhere, I suggest you modify the `Builder::cargo` function (in `builder.rs`) such that `rust_polly_tests` is used when `cmd` is "test" or "bench".

Also, why the `rust_polly_tests` is used (`is_test == true`) for "check" and "doc" in the current implementation?

Reply:
> Instead of modifying rustc_cargo_env, which forces one to pass a magic boolean everywhere, I suggest you modify the Builder::cargo function (in builder.rs) such that rust_polly_tests is used when cmd is "test" or "bench".

Ok, that sounds better.

> Also, why the rust_polly_tests is used (is_test == true) for "check" and "doc" in the current implementation?

I want to make sure all tests are run with the polly optimization, to guard against possible polly bugs.

Reply:
> I want to make sure all tests are run with the polly optimization, to guard against possible polly bugs.

No tests would be run with `cargo check` and `cargo doc` though. Maybe "test" isn't a good name for this configuration 🤔.